clients/java/signalr/src/main/java/JsonHubProtocol.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.util.ArrayList;
+
+public class JsonHubProtocol implements IHubProtocol{
+    private JsonParser jsonParser = new JsonParser();
+    private static final String RECORD_SEPARATOR = "\u001e";
+
+    @Override
+    public String name() {
+        return "json";
+    }
+
+    @Override
+    public int version() {
+        return 1;
+    }
+
+    @Override
+    public TransferFormat transferFormat() {
+        return TransferFormat.Text;
+    }
+
+    @Override
+    public InvocationMessage[] parseMessages(String payload) {
+
+        String[] messages = payload.split(RECORD_SEPARATOR);
+        ArrayList<InvocationMessage> invocationMessages = new ArrayList<>();
+        // Empty handshake response "{}". We can ignore it
+        for (String splitMessage: messages) {
+            if (splitMessage.length() == 2) {
+                continue;
+            }
+
+            JsonObject jsonMessage = jsonParser.parse(splitMessage).getAsJsonObject();
+            String messageType = jsonMessage.get("type").toString();
```

Comment:
make an enum and parse it back?

Reply:
Yeah, that's a better approach and eventually what I'll do. This is still very early stages. Just started on this this week 😄